src/sqlancer/mysql/oracle/MySQLDQEOracle.java

```diff
@@ -0,0 +1,498 @@
+package sqlancer.mysql.oracle;
+
+import static sqlancer.ComparatorHelper.getResultSetFirstColumnAsString;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.beust.jcommander.Strings;
+
+import sqlancer.Randomly;
+import sqlancer.common.oracle.DQEBase;
+import sqlancer.common.oracle.TestOracle;
+import sqlancer.common.query.SQLQueryAdapter;
+import sqlancer.common.query.SQLQueryError;
+import sqlancer.common.query.SQLancerResultSet;
+import sqlancer.common.schema.AbstractRelationalTable;
+import sqlancer.common.schema.AbstractTable;
+import sqlancer.mysql.MySQLErrors;
+import sqlancer.mysql.MySQLGlobalState;
+import sqlancer.mysql.MySQLSchema;
+import sqlancer.mysql.MySQLSchema.MySQLColumn;
+import sqlancer.mysql.MySQLSchema.MySQLTable;
+import sqlancer.mysql.MySQLSchema.MySQLTables;
+import sqlancer.mysql.MySQLVisitor;
+import sqlancer.mysql.ast.MySQLExpression;
+import sqlancer.mysql.gen.MySQLExpressionGenerator;
+
+public class MySQLDQEOracle extends DQEBase<MySQLGlobalState> implements TestOracle<MySQLGlobalState> {
+
+    private final MySQLSchema schema;
+    private static final String APPEND_ORDER_BY = "%s ORDER BY %s";
+    private static final String APPEND_LIMIT = "%s LIMIT %d";
+    private final List<String> orderColumns = new ArrayList<>();
+    private boolean generateLimit;
+    private boolean generateOrderBy;
+    private boolean operateOnSingleTable;
+    private int limit;
+
+    public MySQLDQEOracle(MySQLGlobalState state) {
+        super(state);
+        schema = state.getSchema();
+
+        MySQLErrors.addExpressionErrors(selectExpectedErrors);
+
+        MySQLErrors.addExpressionErrors(updateExpectedErrors);
+        updateExpectedErrors.add("cannot be null");
```

Comment:
Instead of re-adding the errors here, can we just use existing methods like https://github.com/sqlancer/sqlancer/blob/main/src/sqlancer/mysql/MySQLErrors.java#L60?